integration/Substra/plot_substrafl_flamby.py

```diff
@@ -0,0 +1,446 @@
+from substrafl.dependency import Dependency
+from substrafl.strategies import FedAvg
+from substrafl.nodes import AggregationNode
+from substrafl.evaluation_strategy import EvaluationStrategy
+from substrafl.index_generator import NpIndexGenerator
+from substrafl.experiment import execute_experiment
+
+import torch
+
+
+# ## Substrafl algo definition
+# 
+# To instantiate a Substrafl [**Torch Algorithm**](https://docs.substra.org/en/latest/substrafl_doc/api/algorithms.html#torch-algorithms), you need to define a torch Dataset with a specific `__init__` signature, that must contain (self, x, y, is_inference). This torch Dataset is useful to preprocess your data on the `__getitem__` function.
+# 
+# This dataset is passed **as a class** to the [**Torch Algorithms**](https://docs.substra.org/en/latest/substrafl_doc/api/algorithms.html#torch-algorithms).
+# Indeed, this torch Dataset will be instantiated within the algorithm, using the opener functions as x and y parameters.
+# 
+# The index generator will be used a the batch sampler of the dataset, in order to save the state of the seen samples during the training, as Federated Algorithms have a notion of `num_updates`, which forced the batch sampler of the dataset to be stateful.
+# 
+# `NUM_UPDATES` is the number of model update between each FL strategy aggregation.
+# 
+# Concerning the [**TorchFedAvgAlgo**](https://docs.substra.org/en/latest/substrafl_doc/api/algorithms.html#torchfedavgalgo) interaction with [**FLamby**](https://github.com/owkin/FLamby), we need to overwrite the `_local_predict` function, used to compute the predictions of the model. In the default `_local_predict`provided by **Substrafl**, we assume that the `__getitem__` of the dataset take into account the `is_inference` argument to only return the inputs samples. 
+# 
+# But as the `__getitem__` function is provided by [**FLamby**](https://github.com/owkin/FLamby), the `is_inference` argument is ignored. We need to overwrite the `_local_predict` to change the line:
+# 
+# `for x in predict_loader:` 
+# 
+# to:
+# 
+# `for x, _ in predict_loader:`.
+
+# In[9]:
+
+
+NUM_UPDATES = 16
+SEED = 42
+
+index_generator = NpIndexGenerator(
+    batch_size=BATCH_SIZE,
+    num_updates=NUM_UPDATES,
+)
+ 
+
+class TorchDataset(FedTcgaBrca):
+    
+    def __init__(self, x, y, is_inference):
+        config = x
+        super().__init__(**config)
+
+
+class MyAlgo(TorchFedAvgAlgo):
+    def __init__(self):
+        super().__init__(
+            model=model,
+            criterion=criterion,
+            optimizer=optimizer,
+            index_generator=index_generator,
+            dataset=TorchDataset,
+            seed=SEED,
+        )
+
+    def _local_predict(self, predict_dataset: torch.utils.data.Dataset, predictions_path):
+
+        predict_loader = torch.utils.data.DataLoader(predict_dataset, batch_size=self._index_generator.batch_size)
+
+        self._model.eval()
+
+        predictions = torch.Tensor([])
```

Comment:
predictions should be instantiated as a `(len(predict_loader.dataset), nouttput_dim)` object already not concatenated on the go as this is inefficient.

Reply:
To be investigated but out of scope of the example as it is the substrafl implementation.

Reply:
I don't understand your answer.
The current implementation is terribly inefficient and the modification should take less than two lines.
This has no connexion with the way it is implemented in substrafl, no ?

Reply:
To be more precise the user is invited to overwrite the local predict method of the algo therefore she or he should be able to use efficient implementations as they please.